pkg/setting/setting_unified_alerting.go

```diff
@@ -378,6 +381,11 @@ func (cfg *Cfg) ReadUnifiedAlertingSettings(iniFile *ini.File) error {
 		return err
 	}
 
+	uaCfg.NotificationLogRetention, err = gtime.ParseDuration(valueAsString(ua, "notification_log_retention", (5 * 24 * time.Hour).String()))
```

Comment:
Do we need validation to prevent this from being zero?

Reply:
I guess not because I can run Alertmanager with 0s retention too:

```
alertmanager --config.file=config.yml --data.retention=0s
ts=2024-04-04T17:15:15.687Z caller=main.go:184 level=info msg="Starting Alertmanager" version="(version=0.27.0, branch=main, revision=14cbe6301c732658d6fe877ec55ad5b738abcf06)"
...
ts=2024-04-04T17:15:15.718Z caller=main.go:457 level=warn component=configuration msg="repeat_interval is greater than the data retention period. It can lead to notifications being repeated more often than expected." repeat_interval=4h0m0s retention=0s route={}
ts=2024-04-04T17:15:15.718Z caller=main.go:457 level=warn component=configuration msg="repeat_interval is greater than the data retention period. It can lead to notifications being repeated more often than expected." repeat_interval=4h0m0s retention=0s route="{}/{foo=\"bar\"}"
```

Reply:
Zero is a valid value for `-data.retention` in the AM